### ISSUE TITLE
Expand selection with shift+home & shift+end redux

### ISSUE THREAD

**Issue by AdamAndersonFalafelSoftware** [jedmao/vscode-tabsanity#8]
I currently have a state that persists past a restart of VSCode and does the following:
- If I put the cursor in the middle of a line, hold Shift, then press up arrow, end => The cursor jumps to the end of the line where I started holding Shift instead of the end of the current line.
- If I put the cursor in the middle of a line, hold Shift, then press down arrow, home => The cursor jumps to the home of the line where I started holding Shift instead of the home of the current line.

Conversely, these scenarios behave as expected:
- Cursor in middle of line, hold Shift, press up arrow, home => Selection expands from middle of current line to home of current line, as expected
- Cursor in middle of line, hold Shift, press down arrow, end => Selection expands from middle of current line to end of current line, as expected.

In summary, holding Shift and expanding selection to a different line, then trying to **shrink** selection to the home or end of the new line with Home or End keys causes the cursor to jump to Home or End of the starting line. Further expanding selection with Home or End stays on the new line as expected.

![tabsanity](https://cloud.githubusercontent.com/assets/2729807/17623524/7efc753a-606e-11e6-9188-779e9f1e351a.gif)

**Comment by jednano** [comment]
Thanks for reporting this! I can't say when I'll have time to fix this, but I'm willing to accept a PR.

**Comment by fendy3002** [comment]
Hi, I found that it is enough to only remove shift+home and shift+end keybinding from sanity, default back to vscode's keybinding.

As for the workaround, one can add the following setting to keybinding.json:
   ```
 {
        "key": "shift+end",
        "command": "-tabsanity.cursorEndSelect",
        "when": "editorTextFocus"
    },
    {
        "key": "shift+home",
        "command": "-tabsanity.cursorHomeSelect",
        "when": "editorTextFocus"
    }
```

**Comment by krisutofu** [comment]
This bug still is not fixed and much confused me. I installed multiple new VSCode extensions and I found, it's this extension after disabling multiple extensions.
